src/sentry/flags/providers.py

```diff
@@ -48,12 +50,61 @@ class InvalidProvider(Exception):
     ...
 
 
+class LaunchDarklyItemSerializer(serializers.Serializer):
+    accesses = serializers.ListField(required=True)
+    date = serializers.IntegerField(required=True)
+    member = serializers.DictField(required=True)
+    name = serializers.CharField(max_length=100, required=True)
+
+
+"""
+LaunchDarkly has a lot more flag actions than what's in our
+ACTION_MAP. The "updated" action is the catch-all for actions
+that don't fit in the other buckets.
+"""
+
+
+def handle_launchdarkly_actions(action: str) -> int:
+    if action == "createFlag" or action == "cloneFlag":
+        return ACTION_MAP["created"]
+    if action == "deleteFlag":
+        return ACTION_MAP["deleted"]
+    else:
+        return ACTION_MAP["updated"]
```

Comment:
Is every action that's not create, clone, or delete guaranteed to be an update and something we want to track?

Reply:
yup, pretty much everything else is an update -- went through the actions [here](https://docs.launchdarkly.com/home/account/role-actions#feature-flag-actions) and i don't see anything that jumps out as necessary for a new category

Reply:
although, there are probably some actions we don't care as much about (like `reviewApprovalRequest`) but don't know if we should be arbitrarily excluding this/any other actions from the audit log right now

Reply:
I think we should err on the side of excluding immaterial events.  Since debuggability is our primary goal we should drop audit-log events that are not directly involved in altering the program's state.

Reply:
Does it make sense to ingest all events but filter them out at the API level for example?
We could show them all on a dedicated 'flag view log' that shows the complete list.

I'm thinking this because if new flag changes that are relevant to us are added later, we could easily change our API endpoint to include it in the response, for example. And we'd hear from customers (I expect) quickly that "why don't I see this event in the timeline, but I do in the full flag history, that's helpful to debug".

Reply:
> Its not clear to me how a "reviewApprovalRequest" could ever help debug an error.

I agree. But my concern was not about this specific flag. It's about maintenance. If we filter only the keys we believe are valuable today. We'll need to maintain that list as things change on their end and new keys are introduced on their system.

> Keeping a full log of a customer's interactions with a third-party service provider is probably not something Sentry wants to do.

Why not? Please articulate: "Because it's a lot of data", "it could have PII", "security", etc.

> Its too much data with too little utility and too much baggage.
This partially addresses the previous sentence. So the previous one could have been dropped as it didn't add value.

> customers update their SDKs

Except customers don't update their SDKs. We know this at Sentry.

> Which I don't want to do

This language isn't helpful or precise. Pl ease try to stick on 'why' you don't want to. Which you did in the previous paragraph and I only partially disagree. but we won't do or not do something just because "someone wants or not wants to".

Reply:
> I agree. But my concern was not about this specific flag. It's about maintenance. If we filter only the keys we believe are valuable today. We'll need to maintain that list as things change on their end and new keys are introduced on their system.

We can not guard against an un-versioned API changing.  They could change the serialization format to XML, they could rename fields, and they could rename all their actions and this would break.  At some level we are accepting a maintenance burden.  There is no way around it.

When a flag is created, updated, or deleted we want to know about it.  If we allow everything through how should we map `requestReviewApproval` or `billingPaymentSuccess` or `userInvited`?  How should that information be displayed to the user?

> Why not? Please articulate: "Because it's a lot of data", "it could have PII", "security", etc.

All of the above.  But I would add "not useful for the product".  I said as much here: "Its too much data with too little utility and too much baggage."

> Except customers don't update their SDKs. We know this at Sentry.

Not applicable to our situation.  We control when and how the hook handler is updated.

> This language isn't helpful or precise. Pl ease try to stick on 'why' you don't want to. 

I did: "[it] would mean leaking far too much provider specific metadata into our implementation".  We do not want to expose provider specific metadata into our API endpoints because it means our work goes from `1` to `n` where `n` is the number of providers.  Our implementation becomes fragile (provider outages lead to Sentry downtime) and we need the customer to specify (somehow) the provider they're using before querying.

If we were to do the above, my recommendation would be to drop the ingestion webhook and read the audit-log from the provider directly.  I don't personally believe in that strategy but its something we can do.

> Another approach is a denylist. Block off what we know is not valid. But let new things flow through. We can still filte them on our endpoint.

What good does that do us?  Let's say we allow some new action `reviewApprovalRequest` to come through because its not in the denylist.  What action should it map to?  Created, updated, or deleted?  What purpose does it serve the user given our current product positioning?

Reply:
> We can not guard against an un-versioned API changing. They could change the serialization format to XML, they could rename fields, and they could rename all their actions and this would break. At some level we are accepting a maintenance burden. There is no way around it.

I agree we can't guard about unversioned API changes. But I also didn't talk about that at all, only about new 'reasons' being added.

> Not applicable to our situation. We control when and how the hook handler is updated.

Right, that makes sense. When you mentioned "customers update SDK" I felt the need to comment on it. 

> What good does that do us? Let's say we allow some new action reviewApprovalRequest to come through because its not in the denylist. What action should it map to? Created, updated, or deleted? What purpose does it serve the user given our current product positioning?

We could add to the denylist. But if we got a flag that was useful, we could map to something that shows up to our users.
Of course if it turns out we're just adding stuff to the denylist it defeated the purpose of low maintenance I was trying to raise.

I'm happy to take your approach on this.